File: NewStart/stringPalindrome/eg1.cpp
#include<iostream>
using namespace std;
bool check(const string &s,int i,int j)
{
while(i<j)
{
if(s[i]!=s[j]) return false;
i++;
j--;
}
return true;
}
int main()
{
int start,end,maxi,len;
string ans;
maxi=0;
string s="abarrmalayalamrraab";
for(int i=0;i<s.size();i++)
{
for(int j=i;j<s.size();j++)
{
start=i;
end=j;
if(check(s,start,end))
{
len=end-start+1;
cout<<"Palindromic Substring = "<<s.substr(start,len)<<endl;
if(len>maxi)
{
maxi=len;
ans=s.substr(start,len);
}
}
}
}
cout<<"Longest Palindromic sub string size - "<<maxi<<endl;
cout<<"Longest Palindromic sub string - "<<ans<<endl;

return 0;
}

File: JAVA_CODE/palindorme/eg1.java
class psp
{
public static boolean isPalindrome(String s,int e,int f)
{
while(e<f)
{
if(s.charAt(e)!=s.charAt(f)) return false;
e++;
f--;
}
return true;
}
public static void main(String gg[])
{
String s="abarrmalayalamrraab";
//Generate all sub strings 
System.out.println("Printing All Sub strings");
int maxi=0;
int start,end;
start=end=0;
for(int i=0;i<s.length();i++)
{
for(int j=i;j<s.length();j++)
{
if(isPalindrome(s,i,j)) 
{
if(j-i+1>maxi)
{
maxi=j-i+1;
start=i;
end=j;
}
System.out.print(s.substring(i,j+1)+" ");
}
}
}
System.out.println("\n LOngest Plaindromic Substring is of legnth - "+maxi+" : ("+s.substring(start,end+1)+")");
}
}